### PR TITLE
New CI pipeline to test autorest.python for azure-core

### PR DESCRIPTION
**Pull request by praveenkuttappan**
New pipeline yml to test autorest.python for azure-core whenever azure-core changes.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: .azure-pipelines/autorest_checks.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+variables:
+  NodeVersion: '10.x'
+  PythonVersion: '3.6'
+  auto_rest_clone_url: 'https://github.com/Azure/autorest.python.git'
+  repo_branch: 'azure-core'
+  source_path: 'sdk/core/azure-core'
+
+jobs:
+  - job: 'Run_AutoRest'
+    displayName: 'Run AutoRest'
```
Comment (scbedd) [inline .azure-pipelines/autorest_checks.yml]: I didn't know you could do `displayName` for _jobs_! Definitely will be taking advantage of this.

File: .azure-pipelines/autorest_checks.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+trigger:
```
[the remark this answers is not in the document]
Comment (praveenkuttappan) [reply .azure-pipelines/autorest_checks.yml]: I have made this change and moved this yml to eng\pipelines

File: sdk/core/azure-core/dev_requirements.txt [hunk 3]
```diff
@@ -7,3 +7,4 @@ opencensus>=0.6.0
 opencensus-ext-azure>=0.3.1
 opencensus-ext-threading
 mock
+
```
Comment (scbedd) [inline sdk/core/azure-core/dev_requirements.txt]: You can remove this, given that you can't really test immediately.

Comment (scbedd) [reply sdk/core/azure-core/dev_requirements.txt]: Which will also clear your checks and allow you to merge :)

**Pull request merged**